lib/Semantics/tools.cpp

```diff
@@ -956,6 +956,31 @@ bool IsPolymorphicAllocatable(const Symbol &symbol) {
   return IsAllocatable(symbol) && IsPolymorphic(symbol);
 }
 
+static const Scope *FindModuleAncestor(const Symbol &symbol) {
```

Comment:
Might be useful elsewhere; consider adding to `tools.h`.

Reply:
I moved it to tools.h and used it a couple of other places.